src/Portability/Converter.php

```diff
@@ -53,7 +53,7 @@ public function __construct(bool $convertEmptyStringToNull, bool $rightTrimStrin
     /**
      * @param array<int,mixed>|false $row
      *
-     * @return list<mixed>|false
+     * @return non-empty-list<mixed>|false
```

Comment:
The converter methods can't really make that claim. If you feed them with an empty array, you'll get an empty array.

Reply:
@derrabus you mean the same for `convertAssociative`, `convertAllNumeric`, etc. too, right?

Reply:
Looking at it again, feeding an empty array should indeed return an empty array, but the use case would not make sense to me (I might be wrong here!). 

I guess this could be a concern in terms of compatibility, but if you allow feeding an empty array in those cases I think the whole typing proposition here cannot be guaranteed. I'll update the PR in a minute with my proposition 😄

Reply:
@derrabus that would be something like dependent types, which aren't around yet, I think :)

Reply:
You guys are probably going to kill me, but if you are OK with this approach, it seems to work well:

```php
/**
 * @param list<mixed>|false $row
 *
 * @return list<mixed>|false
 * @phpstan-return (
 *     $row is non-empty-list<mixed> ? non-empty-list<mixed> :
 *     ($row is list<mixed> ? list<mixed> : false)
 * )
 */
public function convertNumeric(array|false $row): array|false
{
    return ($this->convertNumeric)($row);
}
```

This typing seems to be compatible with the old signature and is rich enough to get a clean PHPStan/Psalm report.

These tests are also passing too:

```php
/** @return false */
function convertNumericFalse(Converter $converter): bool
{
    return $converter->convertNumeric(false);
}

/** @return list<mixed> */
function convertNumericEmptyArray(Converter $converter): array
{
    return $converter->convertNumeric([]);
}

/** @return non-empty-list<mixed> */
function convertNumericNonEmptyArray(Converter $converter): array
{
    return $converter->convertNumeric(['foo']);
}
```

Reply:
Those annotations won't win a beauty contest, that's for sure. But well, if they work… go for it! 😅

Reply:
Changes ready @derrabus!